### PR TITLE
Update More block description

### PR DESCRIPTION
**Pull request by Soean**
## Description
Updates the description of the More block.
Fixes: #11458

Old: 
`Want to show only an excerpt of this post on your home page? Use this block to define where you want the separation.`
New: 
`Mark the excerpt of this content. Content before this block will be shown in the excerpt on your archives page.`

### CODE REVIEW

**Comment by gziolo** [comment]
It changes more than `home page` to `homepage`. @alexislloyd what's your take on the proposed change? It looks like you iterated on it a bit earlier, so I would expect this PR to be scoped to only fix the way `home page` is used.

**Comment by swissspidy** [comment]
Only fixing `homepage` everywhere in a more scoped fashion makes sense.

We can then improve the text in a separate PR. Could we then get rid of the "post" bit of the description as well please? Otherwise it's not accurate for other post types. I would replace it with "content".

File: packages/block-library/src/more/index.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const name = 'core/more';
 export const settings = {
 	title: _x( 'More', 'block name' ),
 
-	description: __( 'Want to show only an excerpt of this post on your home page? Use this block to define where you want the separation.' ),
+	description: __( 'Do you want to show only an excerpt of this post on your homepage? Then use this block to define where the separation should take place.' ),
```
Comment (tofumatt) [inline packages/block-library/src/more/index.js]: A few thoughts:

1. the leading question only very indirectly explains what this does; we should just outright state what the block does and how it works.
2. should we be saying "post" and "homepage"? Some people don't use the archives listing on their homepage and Gutenberg edits more than just post content.

Here's my suggestion:

```suggestion
	description: __( 'Do you want to show only an excerpt of this post on your homepage? Then use this block to define where the separation should take place.' ),
	description: __( 'Mark the excerpt of this content. Content before this block will be shown in the excerpt on your archives page.' ),
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
I don't like block descriptions that lead with questions. We use an entire sentence to ask the user if this is the block they want to be using before telling them what it does; the question is an indirect way of asking them and I don't think it's very instructive 😄

File: packages/block-library/src/more/index.js (hunk 1 above)
Comment (tofumatt) [reply packages/block-library/src/more/index.js]: I guess we could use "signify" instead of "mark" if that's an unclear verb here. 🤷‍♂️

**Comment by gziolo** [comment]
I updated `home page` with `homepage` with #11661 to land it in 4.3 and resolve the original issue. Feel free to continue discussion on the improvements to this description without the pressure of time :)

**Review by tofumatt — APPROVED** [review]

**Review by tofumatt — APPROVED** [review]
Looks good, I say :shipit:

**Pull request merged**